### PR TITLE
Move Timeout Manager state to the Store

### PR DESCRIPTION
**Pull request by erikdubbelboer**

### CODE REVIEW

File: features/support/types.ts [hunk 1]
```diff
@@ -11,12 +11,14 @@ const allEvents = ['close', 'ready', 'lobby', 'connected', 'disconnected', 'reco
...
   public lastReceivedLobbies: LobbyListEntry[] = []
   public events: RecordedEvent[] = []
+  public scanIndex = 0
 
   constructor (public name: string, public network: Network) {
     allEvents.forEach(eventName => {
       const events = this.events
       this.network.on(eventName as any, function () {
         // console.log(name, `(${network.id})`, 'received event', eventName, `${arguments[0] as string}`, arguments[1], arguments[2])
+        // fs.writeFileSync(`/Users/koen/${name}.log`, `received event ${eventName}(${arguments[0] as string}, ...)\n`)
```
Comment (erikdubbelboer) [inline features/support/types.ts:21]:
```suggestion
```

File: internal/signaling/timeout_manager.go [hunk 2]
```diff
@@ -2,29 +2,27 @@ package signaling
...
-	"github.com/poki/netlib/internal/util"
+	"github.com/poki/netlib/internal/signaling/stores"
 	"go.uber.org/zap"
 )
 
 type timedPeer struct {
 	peer *Peer
 	time time.Time
+	game string
 }
```
Comment (erikdubbelboer) [inline internal/signaling/timeout_manager.go:18]:
```suggestion
```

File: internal/signaling/timeout_manager.go [hunk 3]
```diff
@@ -38,51 +30,57 @@ func (i *TimeoutManager) Run(ctx context.Context) {
...
+			logger.Debug("peer timed out closing peer", zap.String("id", peerID))
+
+			for _, lobby := range lobbies {
+				packet := DisconnectPacket{
+					Type: "disconnect",
+					ID:   peerID,
+				}
+				data, _ := json.Marshal(packet)
+				ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+				defer cancel()
```
Comment (cristaloleg) [inline internal/signaling/timeout_manager.go:44]: Defer in a loop (sorry for the interruption 🙈 )

Comment (koenbollen) [reply internal/signaling/timeout_manager.go:44]: Woops! I completely missed your comment before merging. My bad. Nice catch! Will quickly fix it in a new PR. 👍🏻

Comment (koenbollen) [reply internal/signaling/timeout_manager.go:44]: Here is the fix: https://github.com/poki/netlib/pull/36

Thanks for the interruption! 👍🏻

**Pull request merged**